concreteproperties/design_codes.py

```diff
@@ -727,3 +739,1677 @@ def biaxial_bending_diagram(
             live.refresh()
 
         return f_bb_res, phis
+
+
+class NZS3101(DesignCode):
+    """Design code class for the New Zealand concrete design standard NZS3101:2006. Also
+    implements the requirements of the NZSEE C5 assessment guidelines for probable
+    strength design.
+
+    .. note::
+        Note that this design code currently only supports
+        :class:`~concreteproperties.material.Concrete` and
+        :class:`NZS3101.SteelBarNZ` material objects. Meshed
+        :class:`~concreteproperties.material.Steel` material objects are **not**
+        supported as this falls under the composite structures design code.
+    """
+
+    # TODO - Implement phi=0.75 option for singly reinforced wall design
+
+    def __init__(self):
+        """Inits the NZS3101 class."""
+        self.analysis_code = "NZS3101:2006"
+        super().__init__()
+
+    @dataclass
+    class SteelBarNZ(SteelBar):
+        """Class for a steel bar material to NZS3101, treated as a lumped circular mass
+        with a constant strain.
+
+        :param name: Steel bar material name
+        :param density: Steel bar density (mass per unit volume)
+        :param phi_os: Overstrength factor depending on reinforcement grade
+            (:math:`\phi_{o,f_y}`), refer to NZS3101:2006 CL 2.6.5.5 or NZSEE C5
+            assessment guidelines C5.4.3
+        :param stress_strain_profile: Steel bar stress-strain profile
+        :param colour: Colour of the material for rendering
+        :param meshed: If set to True, the entire material region is meshed; if set to
+            False, the material region is treated as a lumped circular mass at its
+            centroid
+        """
+
+        name: str
+        steel_grade: str
```

Comment:
noted this is an oversight, have added and rather than repeat the longer docstring associated with this I've directed to the `create_steel_material()` method.